nematode/GPSFix.cpp

```diff
@@ -244,21 +242,19 @@ GPSFix::~GPSFix()
 }
 
 // Returns the duration since the Host has received information
-seconds GPSFix::timeSinceLastUpdate()
+milliseconds GPSFix::timeSinceLastUpdate()
 {
-  time_t now      = time(NULL);
-  struct tm stamp = {0};
-
-  stamp.tm_hour = timestamp.hour;
-  stamp.tm_min  = timestamp.min;
-  stamp.tm_sec  = (int)timestamp.sec;
-  stamp.tm_year = timestamp.year - 1900;
-  stamp.tm_mon  = timestamp.month - 1;
-  stamp.tm_mday = timestamp.day;
-
-  time_t then   = mktime(&stamp);
-  uint64_t secs = (uint64_t)difftime(now, then);
-  return seconds((uint64_t)secs);
+  auto time1 = std::chrono::gps_clock::now();
+
+  // GPS epoch is 1980-01-06 00:00:00.00000
+  milliseconds time2 = std::chrono::years(timestamp.year - 1980) +
```

Comment:
We can re-use the `GPSTimestamp::getTime()` if it returns the `sys_time<milliseconds>` `timepoint` rather than a `duration` and then simply calculate the duration by simple subtraction of the `timepoint`s:

```c++
milliseconds GPSFix::timeSinceLastUpdate()
{
  sys_time<milliseconds> ts_tp = timestamp.getTime();
  auto now_tp = system_clock::now();
  return duration_cast<milliseconds>(now_tp - ts_tp);
}
```